### ISSUE TITLE
Missing error for interceptors for dynamic calls

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#70321]
**Version Used**: 
```
Microsoft Visual Studio Professional 2022
Version 17.8.0 Preview 2.0
VisualStudio.17.Preview/17.8.0-pre.2.0+34112.27
Microsoft .NET Framework
Version 4.8.09037
```

**Steps to Reproduce**:

Compile and run the following code:
```cs
using System.Runtime.CompilerServices;
using System;

class C
{
  public static void Main()
  {
    dynamic c = new C();
    c.M();
  }

  public void M() => throw null!;

  [InterceptsLocation("Program.cs", 9, 7)]
  public void Interceptor() => Console.Write(1);
}

namespace System.Runtime.CompilerServices
{
  [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
  public class InterceptsLocationAttribute : Attribute
  {
    public InterceptsLocationAttribute(string path, int line, int character) { }
  }
}
```

**Expected Behavior**:
Either the call `c.M()` is intercepted by `C.Interceptor()` or an error is reported for an invalid interception attempt

**Actual Behavior**:
No errors reproted. The call is not intercepted at runtime and the program crashes in the original `C.M` method that throws an exception

[jcouv update:] Relates to test plan https://github.com/dotnet/roslyn/issues/67421

**Comment by jaredpar** [comment]
This behavior is "By Design". The use of `dynamic` here means that the binding is handled at runtime not compile time. The runtime binder is not being updated to support interceptors.

**Comment by TessenR** [comment]
Why there's no error on the interceptor itself then?
It looks like the only case where an interceptor silently does nothing instead of emitting an error. E.g. it's also by design that you cannot intercept delegate calls, nameof operators, etc, but all these cases produce an error instead of silently ignoring the interceptor
